spec/controllers/target_controller_spec.rb

```diff
@@ -23,10 +24,41 @@
   end
 
   describe 'GET index' do
-    it 'renders a list of targets' do
-      targets = create_list(:target, 2)
-      get :index
-      expect(assigns(:targets)).to match_array(targets)
+    context 'when the user is logged in' do
+      login_user
+
+      it 'renders a list of targets' do
+        targets = create_list(:target, 2)
+        get :index
+        expect(assigns(:targets)).to match_array(targets)
+      end
+    end
+  end
+
+  describe 'DELETE destroy'  do
+    let!(:target) { create(:target) }
+
+    context 'not being logged in' do
+      it 'does not destroy the target' do
+        expect do
+          delete :destroy, params: { id: target.id }
+        end.to_not change(Target, :count)
+      end
+    end
+
+    context 'being logged in' do
+      login_user
+
+      it 'redirects you' do
```

Comment:
If you can change this to a test that really tests the redirection. As you suggested before
`response.should redirect_to contacts_url`

Reply:
Good fix. But change the name to `it 'redirects to new_target_path'`